client/src/components/Sharing/Sharing.vue

```diff
@@ -0,0 +1,593 @@
+                                >
+                                <b-button
+                                    @click="
+                                        setSharing(
+                                            actions.share_with,
+                                            multiselectValues.sharingCandidates.map(({ email }) => email),
+                                            share_option.no_changes
+                                        )
+                                    "
+                                    block
+                                    variant="outline-primary"
+                                >
+                                    Share Anyway
+                                </b-button>
+                                <b-button @click="getModel()" block variant="outline-danger">Cancel </b-button>
+                            </b-card>
+                        </b-col>
+                    </b-row>
+                </b-collapse>
+            </b-card>
+        </div>
+    </div>
+</template>
+
+<script>
+import Vue from "vue";
+import BootstrapVue from "bootstrap-vue";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { faCopy, faEdit, faUserPlus, faUserSlash, faCaretDown, faCaretUp } from "@fortawesome/free-solid-svg-icons";
+import { getAppRoot } from "onload/loadConfig";
+import { getGalaxyInstance } from "app";
+import SlugInput from "components/Common/SlugInput";
+import axios from "axios";
+import Multiselect from "vue-multiselect";
+import { copy } from "utils/clipboard";
+import ConfigProvider from "components/providers/ConfigProvider";
+import CurrentUser from "components/providers/CurrentUser";
+
+Vue.use(BootstrapVue);
+library.add(faCopy, faEdit, faUserPlus, faUserSlash, faCaretDown, faCaretUp);
+const defaultExtra = () => {
+    return {
+        cannot_change: [],
+        can_change: [],
+        can_share: true,
+    };
+};
+export default {
+    components: {
+        ConfigProvider,
+        FontAwesomeIcon,
+        SlugInput,
+        Multiselect,
+        CurrentUser,
+    },
+    props: {
+        id: {
+            type: String,
+            required: true,
+        },
+        pluralName: {
+            type: String,
+            required: true,
+        },
+        modelClass: {
+            type: String,
+            required: true,
+        },
+    },
+    data() {
+        const Galaxy = getGalaxyInstance();
+        return {
+            isCollapseVisible: false,
+            dismissCountDown: 0,
+            charactersThresholdWarning: "Enter at least 3 characters to see suggestions",
+            elementsNotFoundWarning: "No elements found. Consider changing the search query.",
+            ready: false,
+            threeCharactersEntered: true,
+            hasUsername: Galaxy.user.get("username"),
```

Comment:
We want to avoid accessing global Galaxy like this. I think it would be best to accept a user prop and move CurrentUser to an outer component. (Or to break this component into smaller pieces and use the CurrentUser provider as appropriate)